spark-stream-ingest/submit_job.py

```diff
@@ -0,0 +1,162 @@
+"""
+Copyright (c) 2021. Cognitive Scale Inc. All rights reserved.
+
+"""
+import os
+import subprocess
+import yaml
+import sys
+import json
+
+
+def get_runtime_args(config, driver_spec_loc, token):
+    pyspark_args = config['pyspark']
+    options = pyspark_args['options']
+    # os.environ['SPARK_HOME'] = "/Users/bpandey/projects/cortex6/cortex-fabric-examples/spark-stream-ingest/submit/submit/opt/spark"
+    # os.environ['HADOOP_HOME'] 
+    args = [os.environ['SPARK_HOME'] + "/" + pyspark_args['pyspark_bin']]
+    for key in options.keys():
+        val = options[key]
+        if isinstance(val, str):
+            args.append(key)
+            args.append(val)
+        elif isinstance(val, dict):
+            for y in val.keys():
+                s_val = val[y]
+                args.append(key)
+                args.append("{}={}".format(y, s_val))
+    args.append('--conf')
+    args.append(f"spark.kubernetes.driverEnv.CORTEX_TOKEN={token}")
+    args.append('--conf')
+    args.append(f"spark.kubernetes.driver.podTemplateFile={driver_spec_loc}")
+    args.append(pyspark_args['app_location'])
+    return args
+
+
+def replace_template_variables(template, variables):
+    for key in variables.keys():
+        template = template.replace(f"${{{key}}}", variables[key])
+    return template
+
+
+def get_config_file(config_file_loc):
+    with open(config_file_loc) as json_file:
+        return json.load(json_file)
+
+
+def get_driver_template(driver_file_loc):
```

Comment:
some basic error handling here. maybe build a generic exception handling decorator and attach it to all methods?

something like this - https://medium.com/swlh/handling-exceptions-in-python-a-cleaner-way-using-decorators-fae22aa0abec